backend/service/xkcd.go

```diff
@@ -0,0 +1,88 @@
+package service
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"sync"
+	"time"
+	"xkcd/comics/model"
+)
+
+const xkcdURL = "https://xkcd.com/%d/info.0.json"
+
+func GetComic(comicNum int) (*model.Comic, error) {
+    resp, err := http.Get(fmt.Sprintf(xkcdURL, comicNum))
+    if err != nil {
+        return nil, err
+    }
+    defer resp.Body.Close()
+
+    var comic model.Comic
+    if err := json.NewDecoder(resp.Body).Decode(&comic); err != nil {
+        return nil, err
+    }
+
+    return &comic, nil
+}
+
+func GetLatestComicNumber() (int, error) {
+    resp, err := http.Get("https://xkcd.com/info.0.json")
+    if err != nil {
+        return 0, err
+    }
+    defer resp.Body.Close()
+
+    var latest model.LatestComic
+    if err := json.NewDecoder(resp.Body).Decode(&latest); err != nil {
+        return 0, err
+    }
+
+	fmt.Printf("Latest Comic Number is: %d", latest.Num)
```

Comment:
The formatting of the line here looks wrong. What editor are you using? In VSCode, the official go extension also does formatting on save, to enforce code style.